### PR TITLE
Server auth config and google strategy

### PR DESCRIPTION
**Pull request by DavidBronfen**
part of #60

### CODE REVIEW

File: server/src/api/auth/authRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+const router: Router = Router();
+
+// auth login
+router.route("/login")
+    .get((req, res) => {
+        res.send("You are logging in");
+    });
+
+// auth logout
+// TODO - move the logout endpoint to the user route.
```
Comment (ItamarGronich) [inline server/src/api/auth/authRoutes.ts]: Usually, TODO's tend to be forgotten. It's better to turn them into tickets/issues so you'll actually do them.

File: server/src/app.ts [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+import * as mongo from "connect-mongo";
 
-import * as categoryRoutes from "./api/category/categoryRoutes";
-import * as recipeRoutes from "./api/recipe/recipeRoutes";
+import categoryRoutes from "./api/category/categoryRoutes";
+import recipeRoutes from "./api/recipe/recipeRoutes";
+import authRoutes from "./api/auth/authRoutes";
+
+// API keys and Passport configuration
+import * as passportConfig from "./config/passport-config";
```
Comment (ItamarGronich) [inline server/src/app.ts]: This is unused, Remove this.
```suggestion
```
Also, the comment needs to be changed.

File: server/src/config/configService.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import * as dotenv from "dotenv";
+import * as _ from "lodash";
+import Seed from "../util/seed";
+import { IConfigModel } from "./config.model";
+
+let configService: IConfigModel = {
```
Comment (ItamarGronich) [inline server/src/config/configService.ts]: could be named `appConfig`

File: server/src/config/configService.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            }
+        };
+    }
+}
+
+dotenv.config({path: ".env"});
+const envConfig = configDB();
+const envSecret = configSecret();
+configService = _.merge(configService, envConfig, envSecret);
+loadSeed();
```
Comment (ItamarGronich) [inline server/src/config/configService.ts]: Do you really want to load the seed when the server loads?
Also in production?
wouldn't that overwrite data in production?
I think you should create a dev, test and prod files and load them dependant on environment when this `ConfigService` loads. Maybe not required for this PR. I think you could create an issue for this and do this in a future PR.

File: server/src/config/passport-config.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+);
+
+/**
+ * Login Required middleware.
+ */
+export const isAuthenticated = (req: Request, res: Response, next: NextFunction) => {
+    if (req.isAuthenticated()) {
+        return next();
+    }
+    // TODO - redirect user to client login page in order to login first.
```
Comment (ItamarGronich) [inline server/src/config/passport-config.ts]: Also this TODO.

File: server/src/config/configService.ts (hunk 4 above)
Comment (DavidBronfen) [reply server/src/config/configService.ts]: The `loadSeed()` function checks whether to load the seed data or not based on 'seed' property from the appConfig...

**Pull request merged**